src/main/java/com/aws/greengrass/componentmanager/builtins/ArtifactDownloader.java

```diff
@@ -164,7 +170,8 @@ protected long download(InputStream inputStream, MessageDigest messageDigest) th
                 try {
                     artifactFile.write(buffer, 0, readBytes);
                 } catch (IOException e) {
-                    throw new PackageDownloadException(getErrorString("Error writing artifact."), e);
+                    throw new PackageDownloadException(getErrorString("Error writing artifact"), e).withErrorContext(
+                            e.getClass().getSimpleName(), DeploymentErrorCode.IO_WRITE_ERROR);
```

Comment:
it would have been useful to add a method `withErrorContext(Throwable t, ErrorCode c)` so you wouldn't need to write `e.getClass().getSimpleName()` everywhere

Reply:
will refactor it